h5grove/utils.py

```diff
@@ -9,15 +9,20 @@ def attr_metadata(attrId: h5py.h5a.AttrID) -> dict:
     return {"dtype": attrId.dtype.str, "name": attrId.name, "shape": attrId.shape}
 
 
-def get_entity_from_file(h5file: h5py.File, path: str) -> H5pyEntity:
+def get_entity_from_file(
+    h5file: h5py.File, path: str, resolve_links: bool = True
+) -> H5pyEntity:
     if path == "/":
         return h5file[path]
 
     link = h5file.get(path, getlink=True)
     if isinstance(link, h5py.ExternalLink) or isinstance(link, h5py.SoftLink):
-        try:
-            return h5file[path]
-        except (OSError, KeyError):
+        if resolve_links:
+            try:
+                return h5file[path]
+            except (OSError, KeyError):
```

Comment:
Do you want to get the link info in case of an error rather than an error since the link info can be retrieved otherwise now?

Reply:
Or would that be useful/possible to get the link info along with the metadata of the `Group`, `Dataset` anyway through e.g. a "link" in ("hard", "soft", "external")?

Reply:
Good ideas but I would keep this for another PR :+1:

Reply:
OK ! We indeed do not need to care about backward compatibility for now.